buildah-imagestreams/README.md

```diff
@@ -0,0 +1,135 @@
+## Buildah Using Image Streams
+
+The `Task` and `TaskRun` samples here illustrate both how to:
+
+- Have a `Step` in a `Task` use an `ImageStream` reference for its image
+- Have a `Step` push a built image to the OpenShift internal image registry as part of hosting the built image in an `ImageStream`,
+including doing so with the correct certificate so that the push can be performed with TLS verification turned on.
+
+## Preliminary set up
+
+In addition to the actual create of the `Task` and `TaskRun`, the following steps are needed:
+
+- As an administrator to the cluster, create a new `ClusterRole` that allows your `Task` to fetch the internal image registry
+certificate for communicating over TLS.
+
+```bash
+oc apply -f https://raw.githubusercontent.com/openshift/pipelines-catalog/master/buildah-imagestreams/role-image-registry-ca.yaml
+```
+
+- You can then create a `ClusterRoleBinnding` to the service account you will use for your `TaskRun`.  The example here
+uses the `builder` service account.  That service account exists in every OpenShift namespace and will have the 
+necessary authorization secrets to push images to the internal image registry.  If you are using the `OpenShift Pipelines`
+OLM Operator from the OpenShift Operator Hub, you can also use the `pipeline` service account, as it also has the necessary
+authorization secrets.  You can create the `ClusterRoleBinding` with the following command:
+
+```bash
+oc adm policy add-cluster-role-to-user openshift-image-registry-tekton -z builder
+```
+
+- This `ClusterRole` will enable a POC tool we have included in this sample.  The [OpenShift Build Utilities](https://github.com/gabemontero/obu)
+is some helper code around leveraging some of the features associated
+[OpenShift Builds](https://docs.okd.io/latest/builds/understanding-image-builds.html), from within Tekton.  Please do
+provide feedback via GitHub issues on the repo, as we are looking to see if such pieces of function will be useful
+to Tekton users, versus having to duplicate the function in some sort of manual or roll-you-own fashion.  If it does
+prove useful, we'll most likely include it in some official capacity with `OpenShift Pipelines`.  In this particular sample, we 
+are retrieving the certificate for the internal image registry with the utility. 
+
+- You next need to create a couple of `ImageStreams` that will be used by the `Task`.  The first is the `ImageStream` 
+the built image will be pushed to.  Create it with:
+
+```bash
+oc create is test-imagestream-push
+```
+
+- The second `ImageStream` is for hosting the `buildah` image that we'll use for building and pushing images.  There 
+are examples of using `buildah` in this repo without `ImageStreams` [here](https://github.com/tektoncd/catalog/tree/master/buildah).
+For this `ImageStream`, we recommend creating it with the following `oc` command:
+
+```bash
+oc import-image buildah --confirm --from=quay.io/buildah/stable:v1.11.0
```

Comment:
Is there any benefit in importing the buildah image vs referring to the image direclty on quay or rh catalog?

Reply:
I wanted to illustrate using an imagestream ref as the image for a step.

After that, apply your generic "why we use imagestreams" set of answers.

Reply:
See my usage at https://github.com/openshift/pipelines-catalog/blob/60b1b932fa196fcb5f5ac520d421ff47d2637f3e/buildah-imagestreams/task-imagestream-buildah.yaml#L43

Reply:
I realized I may have misinterpreted your ask here @siamaksade  .... do you want me to add a link to openshift doc that explains why one may want to use imagestreams?